src/compile/output/mod.rs

```diff
@@ -484,6 +500,45 @@ pub struct AggregatedCompilerOutput {
     pub build_infos: BTreeMap<Version, RawBuildInfo>,
 }
 
+pub struct ErrorFilter<'a> {
+    error_codes: Cow<'a, [u64]>,
+    ignored_file_paths: Cow<'a, [PathBuf]>,
+}
+
+impl<'a> ErrorFilter<'a> {
+    // Helper function to check if an error code is ignored
+    pub(crate) fn is_code_ignored(&self, code: Option<u64>) -> bool {
+        match code {
+            Some(code) => self.error_codes.contains(&code),
+            None => false,
+        }
+    }
+
+    // Helper function to check if an error's file path is ignored
+    pub(crate) fn is_file_ignored(&self, file_path: &Path) -> bool {
+        self.ignored_file_paths.iter().any(|ignored_path| file_path.starts_with(ignored_path))
+    }
+}
+
+impl<'a> From<(&'a [u64], &'a [PathBuf])> for ErrorFilter<'a> {
```

Comment:
I'd prefer if this was a `fn new` constructor with `impl Into<Cow>`. I find `From` constructors not idiomatic or intuitive

Reply:
same,

main motivation for this was so we don't need to break if providing &[u64] error codes still works.

but the tuple we should actually remove yes